artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/converter/AmqpCoreConverter.java

```diff
@@ -345,6 +347,17 @@ private static ServerJMSMessage processProperties(ServerJMSMessage jms, Properti
             }
          }
          Object correlationID = properties.getCorrelationId();
+
+         if (properties.getMessageId() != null) {
```

Comment:
This seems like it should move up beside the other message id handling at the top of the method. Its strange this happens much later, right after getting the correlation-id value and before the actual subsequent use of that correlation-id value.

Reply:
Yes your are right, it should be moved to the top. I'll do it. thx